### PR TITLE
Variable inlining refactoring

### PR DESCRIPTION
**Pull request by eldritchconundrum**

### CODE REVIEW

File: src/rewriter.fs [hunk 1]
```diff
@@ -537,18 +537,13 @@ let private simplifyStmt = function
...
+            li
+            |> Analyzer.resolve
+            |> Analyzer.markWrites
         Analyzer.markInlinableFunctions li
-        let mapExpr _ e = e
-        let mapStmt = function
-            | Block b as e -> Analyzer.markInlinableVariables b; e
-            | e -> e
-        mapTopLevel (mapEnv mapExpr mapStmt) li |> ignore
+        Analyzer.markInlinableVariables li
```
Comment (laurentlb) [inline src/rewriter.fs:546]: you shadow `li` in this block, then ignore the value, and use the previous `li` on line 548?

**Review by laurentlb — APPROVED** [review]

File: tests/unit/inline.frag [hunk 2]
```diff
@@ -104,3 +104,21 @@ float noinl179(float x) {
 	x = 100.0;
 	return old + x;
 }
+
+// repro for a bug
```
[the remark this answers is not in the document]
Comment (eldritchconundrum) [reply tests/unit/inline.frag:108]: A missed inlining opportunity. Rename the variable in the inner block and inlining will occur. We can fix this corner case later.

File: src/rewriter.fs (hunk 1 above)
Comment (eldritchconundrum) [reply src/rewriter.fs:546]: Yes. This makes explicit the fact that the inlining passes never change the structure of the tree.
They only mark idents, which is consistent with their name.

Comment (laurentlb) [reply src/rewriter.fs:546]: Then don't assign it to li, this is confusing. You can call ignore.

Comment (laurentlb) [reply src/rewriter.fs:546]: or better, `resolve` and `markWrites` should have type unit
